gnd/src/main/java/com/google/android/gnd/repository/ProjectRepository.java

```diff
@@ -133,7 +133,7 @@ public ProjectRepository(
     return projectLoadingState.map(Loadable::value);
   }
 
-  public void onSelectProject(String projectId) {
+  public void activateProject(String projectId) {
```

Comment:
How about adding a `@NonNull` to the parameter and removing `Optional` below? 

I think all the usages of this method are already checking for non-null project ids. So we can simplify it here.

Reply:
`@NonNull` is the default in our current setup, so we don't need to specify it.

The `Optional` isn't being used to check for nulls here.. not sure I follow?

Reply:
`selectProjectEvent` wraps the `projectId` as an Optional. As you already mentioned, the string can't be null. So I was wondering if we can remove the optional now. Line 138